### PR TITLE
some PWADs have STNUM* patches of various width/height (Avactor.wad)

### PR DESCRIPTION
**Pull request by rfomin**
Maybe at this point we should redraw the entire STBAR every tick?

### CODE REVIEW

File: src/st_lib.c [hunk 1]
```diff
@@ -132,17 +131,30 @@ void STlib_drawNum
     num = -num;
   }
 
+  for (i = 1; i <= 9; ++i)
```
Comment (fabiangreffrath) [inline src/st_lib.c:134]: It shouldn't be necessary to recalculate this over and over again during each function call. Let's store the result in static variables, please.

Comment (rfomin) [reply src/st_lib.c:134]: But we use `STlib_drawNum` for big and small fonts. I guess we can add fields to `st_number_t` or just redraw STBAR every tic.

Comment (rfomin) [reply src/st_lib.c]: I simplified it here: https://github.com/fabiangreffrath/woof/pull/691/commits/54deef1f0abba0a4f7d5dbed5c036eb2702a20df No `oldnum` optimization, but we didn't have it until recently.

File: src/st_stuff.c [hunk 2]
```diff
@@ -851,6 +851,14 @@ void ST_drawWidgets(boolean refresh)
   int i;
   int maxammo = plyr->maxammo[weaponinfo[w_ready.data].ammo];
 
+  if (!crispy_hud)
+  {
+    int x = WIDESCREENDELTA;
```
Comment (fabiangreffrath) [inline src/st_stuff.c:856]: Could you please add a comment here about which areas are copied and why?

File: src/st_stuff.c [hunk 3]
```diff
@@ -851,6 +851,14 @@ void ST_drawWidgets(boolean refresh)
   int i;
   int maxammo = plyr->maxammo[weaponinfo[w_ready.data].ammo];
 
+  if (!crispy_hud)
```
Comment (fabiangreffrath) [inline src/st_stuff.c:854]: `crispy_hud` or `st_crispyhud`?

File: src/st_stuff.c (hunk 2 above)
Comment (rfomin) [reply src/st_stuff.c:856]: I suggest copying the entire STBAR and removing the `oldinum` optimization from `STlib_updateMultIcon`. I'm sure there won't be a measurable difference in performance, and who knows what kind of STBAR will be invented in the future.

Comment (fabiangreffrath) [reply src/st_stuff.c:856]: So, if you could show me an elegant solution that works with all current HUDs, that'd be convincing. 😉

Comment (rfomin) [reply src/st_stuff.c:856]: Not sure if it's elegant, but it's simpler 😄

File: src/st_stuff.c (hunk 3 above)
Comment (rfomin) [reply src/st_stuff.c]: Right, fixed in https://github.com/fabiangreffrath/woof/pull/691/commits/05d6989c875356d2410d11c9ce7b9e356dfa165d

**Review by fabiangreffrath — APPROVED** [review]

**Pull request merged**